lidder.go

```diff
@@ -176,87 +176,122 @@ func (defs *defs) exploreDir(dirname string) error {
 
 func (defs *defs) shouldCheck(filename string) bool {
 	// prioritize exclusions over inclusions
-	// matching any means we don't process the file
+	// any exclusion means we do not process the file
 	for _, exclude := range defs.exclude {
 		if exclude.Match([]byte(filename)) {
 			return false
 		}
 	}
-	// matching any means we process the file
+
+	// any inclusion means we process the file
 	for _, include := range defs.include {
 		if include.Match([]byte(filename)) {
 			return true
 		}
 	}
+
 	return false
 }
 
-func main() {
-	if len(os.Args) != 2 && len(os.Args) != 3 {
-		fmt.Println("usage: lidder config.yaml [file]")
-		fmt.Println("  -- If [file] is not specified, defaults to scanning all files from the current directory recursively")
-		os.Exit(1)
-	}
+func fullScanMode(configFilename, baseDir string) bool {
+	results := parseConfig(configFilename)
 
-	config, err := ioutil.ReadFile(os.Args[1])
+	err := results.exploreDir(baseDir)
 	if err != nil {
 		oops(err)
 	}
 
-	results, err := parse(config)
-	if err != nil {
-		oops(err)
+	testFailed := false
+	for _, rule := range results.Rules {
+		shouldNotBeThere, shouldBeThere := rule.Mismatches()
+		if len(shouldNotBeThere) != 0 || len(shouldBeThere) != 0 {
+			testFailed = true
+			fmt.Println(rule.Pattern)
```

Comment:
also considered encapsulating the loop in a checkRules that took in an output function, but did not like the result, made it less clear.